### PR TITLE
testing unexposed ott parameters in generic solvers

### PR DESCRIPTION
**Pull request by ArinaDanilina**
added test_passing_ott_kwargs function in generic sinkhorn tests, part of #544

### CODE REVIEW

**Comment by ArinaDanilina** [comment]
added test_passing_ott_kwargs_linear in generic gw and fgw tests

**Comment by ArinaDanilina** [comment]
added test_passing_ott_kwargs_quadratic in fgw and gw tests, finished #544

File: tests/problems/generic/test_fgw_problem.py [hunk 1]
```diff
@@ -252,3 +253,51 @@ def test_prepare_different_costs(self, adata_time: AnnData):
...
+        problem = GWProblem(adata=adata_space_rotate)
+        problem = problem.prepare(
+            key="batch",
+            policy="sequential",
+            joint_attr="X_pca",
+            x_attr={"attr": "obsm", "key": "spatial"},
+            y_attr={"attr": "obsm", "key": "spatial"},
+        )
+
+        problem = problem.solve(
```
Comment (MUCDK) [inline tests/problems/generic/test_fgw_problem.py:269]: How long does this take locally? 
As we can test this without convergence of the solver, we can set `max_iterations=1`.

**Review by MUCDK — CHANGES_REQUESTED** [review]
see comments

File: tests/problems/generic/test_fgw_problem.py (hunk 1 above)
Comment (ArinaDanilina) [reply tests/problems/generic/test_fgw_problem.py:269]: with max_iterations=1 it takes 1 min

Comment (ArinaDanilina) [reply tests/problems/generic/test_fgw_problem.py:269]: in Sinkhorn it takes 20 sec

**Comment by ArinaDanilina** [comment]
max_iterations=1 speeds it up 1,5 times, in Sinkhorn it now takes 20 sec and the longest test in fgw is 1 min

File: tests/problems/generic/test_fgw_problem.py (hunk 1 above)
Comment (MUCDK) [reply tests/problems/generic/test_fgw_problem.py:269]: ok can you also set the number of inner iterations to a smaller value please? E.g. also 1

**Review by MUCDK — CHANGES_REQUESTED** [review]
see comment

File: tests/problems/generic/test_fgw_problem.py (hunk 1 above)
Comment (ArinaDanilina) [reply tests/problems/generic/test_fgw_problem.py:269]: done, down to 40 sec

File: tests/problems/generic/test_sinkhorn_problem.py [hunk 2]
```diff
@@ -166,3 +167,26 @@ def test_pass_arguments(self, adata_time: AnnData, args_to_check: Mapping[str, A
...
+    @pytest.mark.parametrize(("memory", "refresh"), [(1, 1), (5, 3), (7, 5)])
+    @pytest.mark.parametrize("recenter", [True, False])
+    def test_passing_ott_kwargs(self, adata_time: AnnData, memory: int, refresh: int, recenter: bool):
+        problem = SinkhornProblem(adata=adata_time)
+        problem = problem.prepare(
+            key="time",
+            policy="sequential",
+        )
+
+        problem = problem.solve(
```
Comment (MUCDK) [inline tests/problems/generic/test_sinkhorn_problem.py:180]: inner iterations = 1

**Review by MUCDK — CHANGES_REQUESTED** [review]
see comments

File: tests/problems/generic/test_sinkhorn_problem.py (hunk 2 above)
Comment (MUCDK) [reply tests/problems/generic/test_sinkhorn_problem.py:180]: are you sure this is how to pass inner iterations? I think it's via `linear_solver_kwargs`, but not entirely sure. Please double-check.

**Review by MUCDK — CHANGES_REQUESTED** [review]
see comment

File: tests/problems/generic/test_sinkhorn_problem.py (hunk 2 above)
Comment (ArinaDanilina) [reply tests/problems/generic/test_sinkhorn_problem.py:180]: here it's a SinkhornProblem so inner iterations get passed just like that. In GWProblem I do pass it in ```linear_solver_kwargs={
                "inner_iterations": 1,
                "max_iterations": 1,
            }```

**Review by MUCDK — APPROVED** [review]
LGTM, thanks!

Btw, `inner_iterations` determines how often we log the error in Sinkhorn. So setting this here doesn't have an effect / is unnecessary, but obv. fine.

**Pull request merged**